src/lib/setupContextUtils.js

```diff
@@ -1007,6 +1007,9 @@ function registerPlugins(plugins, context) {
       }
     }
 
+    // Exclude utilities that are known non-classes (e.g. from the blocklist)
+    output = output.filter((cls) => !context.notClassCache.has(cls))
```

Comment:
Nit: can we wrap this with a guard so that we don't loop over all items unnecessarily? 

```suggestion
    if (context.notClassCache.size > 0) {
      output = output.filter((cls) => !context.notClassCache.has(cls))
    }
```